### PR TITLE
[WIP] add syft package

### PR DESCRIPTION
**Pull request by systemshift**
unchecked boxes just means that I am not certain

Checklist
- [ x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [ x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ x] Source is from official source
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x ] Build number is 0
- [ x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by systemshift** [comment]
@conda-forge/help-python need some help with linting.
using `conda smithy recipe-lint` shows the following error: `jinja2.exceptions.TemplateSyntaxError: expected token 'end of print statement', got 'a2'`

File: recipes/syft/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+{% set name = "syft" %}
+{% set version = "0.2.2a2" %}
+
+package:
+  name: {{ syft }}
```
Comment (npavlovikj) [inline recipes/syft/meta.yaml]: Please use:
```
name: {{ name|lower }}
version: {{ version }}
```
instead.

Comment (npavlovikj) [reply recipes/syft/meta.yaml]: You can read more about using variables here, https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#package-section.

File: recipes/syft/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  version: {{ 0.2.2a2 }}
+
+source:
+  url: https://pypi.org/project/syft/0.2.0a2/
```
Comment (npavlovikj) [inline recipes/syft/meta.yaml]: You need the full path to the tarball here, not the directory (e.g. https://files.pythonhosted.org/packages/97/ba/c01bb914f395611034bab10cb8d1d22e49691534d30f85efbd82f7bd27cf/syft-0.2.0a2.tar.gz).

File: recipes/syft/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+source:
+  url: https://files.pythonhosted.org/packages/97/ba/c01bb914f395611034bab10cb8d1d22e49691534d30f85efbd82f7bd27cf/syft-0.2.0a2.tar.gz
+  sha256: f3839e5be537c3d9b3f530778e9f08cdeeab99d9ba71f85c35b158a2264c7fc6
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
```
Comment (systemshift) [inline recipes/syft/meta.yaml]: @npavlovikj should I replace script with `python setup.py` given the way the repo I am adding is currently structured?

File: recipes/syft/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    - python
+    - pip
+  run:
+    - python
+
+test:
+  imports:
+  # work on later
+    - simplejson
+    - simplejson.tests
```
Comment (systemshift) [inline recipes/syft/meta.yaml]: What should I replace tests with exactly

**Comment by npavlovikj** [comment]
@systemshift , I think there is an issue with `setup.py`. If you look at it, you will see:
```
REQ_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), "pip-dep")
core_reqs, core_dependency_links = get_requirements(os.path.join(REQ_DIR, "requirements.txt"))
```
and if you extract the "syft" tarball, you will see that there is no directory "pip-dep" and no "requirements.txt" file.

Looks like these are all part of the repo, so you may want to check with the developers and ask them to add release for "0.2.2a2" there. You can also comment out the failing lines in "setup.py", but I don't know how that affects the functionality of "syft".

**Comment by systemshift** [comment]
@npavlovikj  So now the issue has been solved by fixing it upstream. A new error has been showing up.

**Comment by npavlovikj** [comment]
I think you should add `pytest-runner` as a `host` dependency.

File: recipes/syft/meta.yaml (hunk 4 above)
Comment (systemshift) [reply recipes/syft/meta.yaml:29]: @npavlovikj currently the build breaks at testing. I need to replace the default tests template with pysyft ones. But not sure exactly with what.

File: recipes/syft/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    - lz4~=3.0.2
+    - msgpack~=1.0.0
+    - numpy~=1.18.1
+    - phe~=1.4.0
+    - Pillow~=6.2.2
+    - requests~=2.22.0
+    - scipy~=1.4.1
+    - syft-proto~=0.4.1
+    - tblib~=1.6.0
+    - torchvision~=0.5.0
```
Comment (xhochy) [inline recipes/syft/meta.yaml:37]: Use different requirements annotation
```suggestion
    - torchvision 0.5.0
```

**Pull request closed without merging**